releasenotes.txt

```diff
@@ -1,4 +1,257 @@
+        * The counter id generator may generate conflicting ids for as much as twelve hours after upgrade.
+        * `update` and `delete` statements will now take into account any enabled filter on the entities
+          they update or delete, while previously they were ignoring them. (`insert` statements will also take
+          them into account, but previously they were failing instead of ignoring enabled filters.)
+        * ISession.Persist will no more trigger immediate generation of identifier.
+        * Bags will no more be loaded with "null" entities, they will be filtered out.
+        * Setting the value of an unitialized lazy property will no more trigger loading of all the lazy
+          properties of the entity.
+        * If an unitialized lazy property has got its value set, without any other subsequent lazy property
+          load on the entity, a dynamic update will occur on flush, even if the entity has dynamic updates
+          disabled. This update will occur even if the set value is identical to the currently persisted
+          property value.
+        * Assigning an uninitialized proxy to a `no-proxy` property will no more trigger the proxy
+          initialization. Moreover, reading the property afterwards will no more unwrap the assigned proxy,
+          but will yield it.
+        * A class having an explicitly implemented interface declaring a member with the same name than the
+          class id will have its proxies trigger a lazy load if the interface "id" is accessed.
+        * SQLite: in order to avoid a floating point division bug losing the fractional part, decimal are now
+          stored as `REAL` instead of `NUMERIC`. Both are binary floating point types, excepted that `NUMERIC`
+          stores integral values as `INTEGER`. This change may cause big integral decimal values to lose more
+          precision in SQLite.
+        * Custom dialects used for databases that do not support cross join will have to override
+          `SupportsCrossJoin` property and set it to `false`.
+        * `VisitorParameters.ConstantToParameterMap` may contain the same parameter for multiple constant
+          expressions.
+        * `ICache` caches yielded by the session factory will be `CacheBase` wrappers around the cache actually
+          provided by the cache provider, if it was not deriving from `CacheBase`.
+
+** Bug
+
+   * #2380 OData NotSupportedException MemberInit on base class member
+   * #2346 Fix SQLite typing
+   * #2336 Intermittent null reference exception on CloseConnection
+   * #2324 Update IIsEntityDecider to use ExpressionsHelper.TryGetMappedType
+   * #2319 Upgrade AsyncGenerator to 0.18.1
+   * #2299 Proper query plan caching for DML LINQ queries
+   * #2286 Wrong sql if used joined-subclass with filters for key columns
+   * #2278 IInterceptor.OnPrepareStatement results not used in insert/update commands
+   * #2266 Fix comment for Restrictions.IsEmpty
+   * #2255 Fix a flaky test
```

Comment:
Is it manually grouping by tasks? Or somehow automatic (like Labels based)? This one seems should go to Tests. Also I see few other test related tasks in Improvements (by search test) - I think they should also go to Tests.

Reply:
@bahusoid automatic

Reply:
And based on `t:` labels.

Reply:
So should we change `t:` labels for test related tasks like this one? As it's not very useful fix for end user.

Reply:
It is automatic for first generation, after that I manually update it, because the tool adds more than what we need and there is some cleanup to do.

Anyway if editing the release notes for changing the category of an issue, it is better to keep the issue typed accordingly.

I do not really mind where these test related tasks are categorized, provided it is not wrong.